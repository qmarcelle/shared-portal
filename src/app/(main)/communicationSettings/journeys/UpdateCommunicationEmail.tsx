import { ChangeAuthDeviceSlide } from '@/components/composite/ChangeAuthDeviceSlide';
import { SuccessSlide } from '@/components/composite/SuccessSlide';
import { AppLink } from '@/components/foundation/AppLink';
import {
    ModalChildProps,
    useAppModalStore,
} from '@/components/foundation/AppModal';
import { Column } from '@/components/foundation/Column';
import { TextField } from '@/components/foundation/TextField';
import { useState } from 'react';
import { InputModalSlide } from '../../../../components/composite/InputModalSlide';
import { Spacer } from '../../../../components/foundation/Spacer';
import { TextBox } from '../../../../components/foundation/TextBox';

const headerText = 'Confirm Email Address';
interface UpdateCommunicationEmailProps {
  email: string;
}

export const UpdateCommunicationEmail = ({
  changePage,
  pageIndex,
  email,
}: ModalChildProps & UpdateCommunicationEmailProps) => {
  const { dismissModal } = useAppModalStore();
  /* const initChange = () => {
    changePage!(1, true);
  };
  function changePageIndex(index: number, showback = true) {
    changePage?.(index, showback);
  } */
  const [mainAuthDevice, setMainAuthDevice] = useState(email);
  const [newAuthDevice, setNewAuthDevice] = useState('');
  const [confirmCode, setConfirmCode] = useState('');

  const initNewDevice = async () => {
    // Do API call for new device
    setMainAuthDevice(newAuthDevice);
    changePage?.(1, true);
  };

  const submitCode = async () => {
    // Do API call for submit code
    changePage?.(2, false);
  };

  const pages = [
    <ChangeAuthDeviceSlide
      key={0}
      label="Update Email Address"
      subLabel="Enter the new email address you'd like to use for communications and security settings."
      actionArea={
        <TextField
          valueCallback={(val) => setNewAuthDevice(val)}
          label="Email Address"
        />
      }
      cancelCallback={() => dismissModal()}
      nextCallback={
        newAuthDevice.length > 5 ? () => initNewDevice() : undefined
      }
    />,
    <InputModalSlide
      key={1}
      label={headerText}
      subLabel="We just need to verify your email. Enter the security code we sent to:"
      actionArea={
        <Column>
          <TextBox
            className="body-1 text-center font-bold"
            text={mainAuthDevice}
          />
          <Spacer size={32} />
          <TextField
            type="text"
            valueCallback={(val) => setConfirmCode(val)}
            label="Enter Security Code"
          ></TextField>
          <Spacer size={16} />
          <AppLink className="self-start" label="Resend Code" />
          <Spacer size={32} />
        </Column>
      }
      cancelCallback={() => dismissModal()}
      nextCallback={confirmCode.length > 5 ? () => submitCode() : undefined}
    />,

    <SuccessSlide
      key={2}
      label="Email Address Updated"
      body={
        <Column className="items-center">
          <TextBox className="text-center" text="Your email address is:" />
          <Spacer size={16} />
          <TextBox text={mainAuthDevice} />
        </Column>
      }
      doneCallBack={() => dismissModal()}
    />,
  ];

  return pages[pageIndex!];
};
