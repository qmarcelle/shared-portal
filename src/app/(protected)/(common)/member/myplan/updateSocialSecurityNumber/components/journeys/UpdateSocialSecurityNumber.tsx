import { ErrorDisplaySlide } from '@/components/composite/ErrorDisplaySlide';
import { InputModalSlide } from '@/components/composite/InputModalSlide';
import { SuccessSlide } from '@/components/composite/SuccessSlide';
import {
  ModalChildProps,
  useAppModalStore,
} from '@/components/foundation/AppModal';
import { Column } from '@/components/foundation/Column';
import { Spacer } from '@/components/foundation/Spacer';
import { TextBox } from '@/components/foundation/TextBox';
import { TextField } from '@/components/foundation/TextField';
import { useEffect, useRef, useState } from 'react';
import { updateSSNData } from '../../actions/updateSSNData';
import { UpdateSSNRequest } from '../../models/app/updateSSNRequest';

interface UpdateSocialSecurityNumberJourneyProps {
  memberName: string;
  successCallback: () => void;
}

const headerText = 'Update or Add a Social Security Number';

export const UpdateSocialSecurityNumberJourney = ({
  changePage,
  pageIndex,
  memberName,
  successCallback,
}: ModalChildProps & UpdateSocialSecurityNumberJourneyProps) => {
  const { dismissModal } = useAppModalStore();
  const ssnInputRef = useRef<HTMLInputElement>(null);
  const [securityNumber, setSecurityNumber] = useState('***-**-****');

  useEffect(() => {
    if (pageIndex === 0) {
      // Focus SSN input on first page
      ssnInputRef.current?.focus();
    }
  }, [pageIndex]);

  const addSecurityNumber = (value: string) => {
    if (!isNaN(Number(value))) {
      setSecurityNumber(value);
    }
  };

  function changePageIndex(index: number, showback = true) {
    changePage?.(index, showback);
  }

  const submitSSNCode = async () => {
    try {
      const request: UpdateSSNRequest = {
        ssn: securityNumber,
      };
      const response = await updateSSNData(request);
      if (response.data?.message === 'Updated SSN') {
        successCallback();
        changePageIndex?.(1, false);
      }
    } catch (errorMessage: unknown) {
      changePageIndex?.(2, false);
    }
  };

  const pages = [
    <InputModalSlide
      key={0}
      label={headerText}
      subLabel="Enter the social security number for:"
      actionArea={
        <Column>
          <TextBox
            className="font-bold text-center"
            text={memberName}
            ariaLabel={`Enter SSN for ${memberName}`}
          />
          <Spacer size={32} />
          <TextField
            ref={ssnInputRef}
            hint="***-**-****"
            valueCallback={(val) => addSecurityNumber(val)}
            maxLength={9}
            label="Social Security Number"
            ariaLabel="Enter your social security number"
            ariaDescribedBy="ssn-description"
            type="password"
          />
          <TextBox
            id="ssn-description"
            className="sr-only"
            text="Enter your 9-digit social security number. The number will be masked for security."
          />
          <Spacer size={32} />
        </Column>
      }
      nextCallback={
        securityNumber.length > 5 ? () => submitSSNCode() : undefined
      }
      cancelCallback={() => dismissModal()}
      buttonLabel="Save Changes"
    />,

    <SuccessSlide
      key={1}
      label="Social Security Number Updated"
      body={
        <Column className="items-center">
          <TextBox
            className="text-center"
            text="Your social security number has been successfully updated."
            ariaLabel="Your social security number has been successfully updated."
          />
          <Spacer size={32} />
        </Column>
      }
      doneCallBack={() => dismissModal()}
    />,
    <ErrorDisplaySlide
      key={2}
      label="Something went wrong."
      body={
        <Column className="items-center">
          <TextBox
            className="text-center"
            text="We're unable to update your information at this time. Please try again later."
            ariaLabel="We're unable to update your information at this time. Please try again later."
          />
        </Column>
      }
      doneCallBack={() => dismissModal()}
    />,
  ];

  return pages[pageIndex!];
};
