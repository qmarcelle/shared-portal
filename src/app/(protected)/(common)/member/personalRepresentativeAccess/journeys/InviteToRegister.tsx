import { InputModalSlide } from '@/components/composite/InputModalSlide';
import { SuccessSlide } from '@/components/composite/SuccessSlide';
import {
  ModalChildProps,
  useAppModalStore,
} from '@/components/foundation/AppModal';
import { Column } from '@/components/foundation/Column';
import { Spacer } from '@/components/foundation/Spacer';
import { TextBox } from '@/components/foundation/TextBox';
import { TextField } from '@/components/foundation/TextField';

interface InviteToRegisterProps {
  memberName: string;
}

export const InviteToRegister = ({
  changePage,
  pageIndex,
  memberName,
}: ModalChildProps & InviteToRegisterProps) => {
  const { dismissModal } = useAppModalStore();
  const initChange = () => {
    changePage!(1, false);
  };

  const pages = [
    <InputModalSlide
      key={0}
      label="Invite to Register"
      subLabel="You're inviting:"
      buttonLabel="Send Invite"
      actionArea={
        <Column>
          <TextBox className="body-1 text-center font-bold" text={memberName} />
          <Spacer size={32} />
          <TextBox
            className="text-center"
            text="We'll send an email inviting them to register an account. We just need their email."
          />
          <Spacer size={32} />
          <TextField type="text" label="Their Email Address"></TextField>
          <Spacer size={32} />
          <TextField
            type="text"
            label="Confirm Their Email Address"
          ></TextField>
          <Spacer size={32} />
        </Column>
      }
      cancelCallback={() => dismissModal()}
      nextCallback={initChange}
    />,
    <SuccessSlide
      key={1}
      label="Invitation to Register Sent"
      body={
        <Column className="items-center ">
          <TextBox
            className="text-center"
            text="We've sent an email inviting this member to register. Once they've completed registration, you'll be able to request access to their information"
          />
        </Column>
      }
      doneCallBack={() => dismissModal()}
    />,
  ];

  return pages[pageIndex!];
};
