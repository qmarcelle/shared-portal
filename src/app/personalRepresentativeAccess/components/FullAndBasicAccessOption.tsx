import { Checkbox } from '@/components/foundation/Checkbox';
import { Column } from '@/components/foundation/Column';
import { Divider } from '@/components/foundation/Divider';
import { Row } from '@/components/foundation/Row';
import { Spacer } from '@/components/foundation/Spacer';
import { TextField } from '@/components/foundation/TextField';

const FullAccessOption = ({ selectedData }: { selectedData: boolean }) => {
  return (
    <main className="">
      {selectedData && (
        <Column className="items-center">
          <Row className="body-1 text-center">
            Grant Basic Access to Chris Hall
          </Row>
          <Spacer size={8} />
          <Divider></Divider>
          <Spacer size={16} />
          <Checkbox
            label=""
            body="I Agree. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            Sed sodales, orci vitae ullamcorper dignissim, 
            ligula nulla condimentum lectus, feugiat interdum dui turpis eget nisl. 
            Duis at lectus vitae augue pulvinar dictum eu nec neque. Pellentesque maximus mattis justo id tincidunt.
             Mauris porttitor"
            selected={true}
          ></Checkbox>
          <Spacer size={32} />
          <TextField
            type="text"
            label="Type Your Full Name"
            value="[Mature Minor Name]"
          ></TextField>
          <Spacer size={32} />
        </Column>
      )}
      {!selectedData && (
        <Column className="items-center">
          <Row className="body-1 text-center">
            Grant Full Access to Chris Hall
          </Row>
          <Spacer size={16} />
          <Divider></Divider>
          <Spacer size={32} />
          <Row className="font-bold">
            HIPAA Authorization for Full Portal Access and Interoperability APIs
          </Row>
          <Spacer size={16} />
          <Row>
            You are asking BlueCross BlueShield of Tennessee to give your
            Personal Representative Full Access to all of your health
            information available in your member portal and through our
            interoperability application programming interfaces (APIs),
            including our Patient Access API and Payer-to-Payer API. This means
            that your Personal Representative will be able to access highly
            sensitive health information about you and to use our APIs to share
            your health information with other third parties through our APIs,
            such as any third-party application of your Personal
            Representative’s choosing or your other health plans.
          </Row>
          <Spacer size={16} />
          <Row>
            By clicking the “I Agree” button after reading this, you authorize
            BlueCross BlueShield of Tennessee to share the following health
            information about you with your Personal Representative and any
            people, companies or other third parties authorized by your Personal
            Representative, if we maintain it:
          </Row>
          <Spacer size={16} />
          <Row>
            <ul>
              <li>
                Demographic and contact information related to you and your
                family;
              </li>
              <li>
                Your claims and encounter data, including cost information; and
              </li>
              <li>
                Certain clinical data, such as allergies and intolerances,
                assessments and plans of treatment, care team members, clinical
                notes, clinical tests, diagnostic imaging, encounter
                information, goals, health insurance information, health status
                assessments, immunizations, laboratory tests and results,
                medications, problems, procedures, provenance, unique device
                identifiers for implantable devices, and vital signs.
              </li>
            </ul>
          </Row>
          <Row>
            The information they will have access to may reveal highly sensitive
            health information about you, including information about your
            treatment or care for reproductive health (e.g., family planning,
            contraception, miscarriage, abortion, maternity, infertility),
            substance use disorders (e.g., drugs and alcohol), mental or
            behavioral health disorders, HIV/AIDS, sexually transmitted diseases
            (STDs), communicable diseases, developmental or intellectual
            disabilities, genetic disorders, (including genetic testing for such
            disorders and genetic history), abuse (e.g., sexual, physical or
            mental), brain or other cognitive disorders, or other sensitive
            information. By clicking on the “I AGREE” button you are
            specifically authorizing the disclosure of this highly sensitive
            health information.
          </Row>
          <Row>
            We are making this disclosure at your request, with your approval
            and at your direction. The intended purposes of the use and
            disclosure of your health information, including your highly
            sensitive health information, are to:
          </Row>
          <Row>
            Give you control over the use and redisclosure of your health
            information, including highly sensitive health information;
          </Row>
          <Row>
            Give your Personal Representative the ability to use our
            interoperability APIs to disclose your highly sensitive health
            information to other people, companies and third parties authorized
            by your Personal Representative without needing further
            authorization from you to do so; and{' '}
          </Row>
          <Row>
            Support treatment, payment and certain health care operations
            activities, including care coordination and case management among
            your care team, including family members, friends and others
            involved in your care.{' '}
          </Row>
          <Row>
            You acknowledge that your Personal Representative and the people,
            companies and other third parties that they authorize to receive
            your health information may not be subject to state or federal
            health information privacy laws, such as the federal Health
            Information Portability and Accountability Act and its implementing
            regulations (collectively, called “HIPAA”), 42 U.S.C. § 290dd-2 and
            its implementing regulations at 42 CFR Part 2 (collectively, called
            “Part 2”) for certain substance use disorder records, or state
            health information confidentiality laws. If your Personal
            Representative or someone that they have authorized to have access
            to your Part 2 substance use disorder records is a HIPAA covered
            entity or business associate (collectively, called a
            “HIPAA-regulated entity”) that is receiving your records to support
            treatment, payment or health care operations activities, the
            HIPAA-regulated entity recipient may redisclose those substance use
            disorder records for any purpose permitted by HIPAA, except for uses
            and disclosures for civil, criminal, administrative, and legislative
            proceedings against you. This authorization for the release of your
            health information is voluntary. This authorization will remain in
            effect until one of the following events happen: It is revoked
            (taken back).{' '}
          </Row>
          <Row>
            XX calendar days after the last time you access your portal account.
            You understand that every time you access your portal account, you
            reaffirm this authorization and the xx-day clock starts again. If
            you do not access your portal account for more than xx-days, you
            will be asked to agree to a new authorization.
          </Row>{' '}
          <Row>
            We reasonably determine that allowing the user you have designated
            as your Personal Representative to link to your account through our
            portal and APIs would present an unacceptable security risk.{' '}
          </Row>
          <Row>
            As otherwise provided for in this authorization or our User
            Agreement, such as if we learn that account credentials have been
            misused.{' '}
          </Row>
          <Row>
            You understand that BlueCross BlueShield of Tennessee will not
            condition the provision of treatment, payment for treatment,
            enrollment in a health plan, or eligibility for benefits on your
            agreement to this authorization. You may take back this
            authorization in writing at any time, except to the extent the
            information has already been disclosed in reliance on this
            authorization. You may take back this authorization by sending a
            request to us the Contact Us page on bcbst.com or by sending us your
            written cancellation to BlueCross BlueShield of Tennessee, Privacy;
            1 Cameron Hill Circle; Building 1, 5th Floor; Chattanooga, Tennessee
            37402. If you decide to take back this authorization, it will not
            affect any disclosure BlueCross BlueShield made in reliance on this
            authorization before the cancellation.{' '}
          </Row>
          <Row>
            {' '}
            Please note that your Personal Representative will not have Full
            Access to your health information or the ability to use our APIs to
            disclose your information to other people, companies or third
            parties if you do not sign this authorization. If you do not sign
            this authorization, your Personal Representative will continue to
            have Basic Access to your health information in accordance with the
            laws that apply to BlueCross BlueShield of Tennessee.{' '}
          </Row>
          <Row>
            Notice To Recipient of Substance Use Disorder Records: 42 CFR Part 2
            prohibits unauthorized use or disclosure of these records.{' '}
          </Row>
          <Spacer size={16} />
          <Divider></Divider>
          <Spacer size={16} />
          <Checkbox
            label=""
            body="I Agree. Clicking “I Agree” means that you have read, understand, and
          agree to this authorization. You are giving your permission for
          BlueCross BlueShield of Tennessee to give your Personal
          Representative, and any person, company or other third party
          authorized by your Personal Representative, access to your health
          information, including your highly sensitive health information. You
          understand that you can contact BlueCross BlueShield of Tennessee for
          a copy of this authorization."
            selected={true}
          ></Checkbox>
          <Spacer size={32} />
          <TextField type="text" label="Type Your Full Name"></TextField>
          <Spacer size={32} />
        </Column>
      )}
    </main>
  );
};

export default FullAccessOption;
