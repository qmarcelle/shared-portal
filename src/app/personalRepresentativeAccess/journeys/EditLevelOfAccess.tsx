import { InitModalSlide } from '@/components/composite/InitModalSlide';
import { InputModalSlide } from '@/components/composite/InputModalSlide';
import { SuccessSlide } from '@/components/composite/SuccessSlide';
import {
  ModalChildProps,
  useAppModalStore,
} from '@/components/foundation/AppModal';
import { Column } from '@/components/foundation/Column';
import { Radio } from '@/components/foundation/Radio';
import { Spacer } from '@/components/foundation/Spacer';
import { TextBox } from '@/components/foundation/TextBox';
import { useState } from 'react';
import FullAndBasicAccessOption from '../components/FullAndBasicAccessOption';

const bottomNote =
  'By sending the code I agree to receive a one-time security code. Message and data rates may apply, Subject to terms and condition.';

interface InviteToRegisterProps {
  memberName: string;
}

export const EditLevelOfAccess = ({
  changePage,
  pageIndex,
  memberName,
}: ModalChildProps & InviteToRegisterProps) => {
  const [selectedData, setSelectedData] = useState(false);

  function handleClick() {
    setSelectedData(true);
    setSelectedData(!selectedData);
  }

  const { dismissModal } = useAppModalStore();

  const pages = [
    <InitModalSlide
      key="first"
      label="Edit Level Of Access"
      subLabel={
        <Column>
          <TextBox className="text-center" text="You're changing access for:" />
          <Spacer size={24} />
          <TextBox className="text-center" text="Chris Hall" />
          <Spacer size={24} />
          <Column>
            <Radio
              label="Full Access"
              subLabel="Your Representative will have access to all documents and claims, even those with sensitive information"
              selected={!selectedData}
              callback={handleClick}
            />
            <Radio
              label="Basic Access"
              subLabel="Your Representative will have access to all documents and claims, but will not be able to view sensitive information"
              selected={selectedData}
              callback={handleClick}
            />
          </Column>
        </Column>
      }
      changeAuthButton={undefined}
      buttonLabel="Next"
      nextCallback={() => changePage?.(1, true)}
      bottomNote={<TextBox className="body-2" text={bottomNote} />}
      cancelCallback={() => dismissModal()}
    />,
    <InputModalSlide
      key="second"
      label="Edit Level Of Access"
      subLabel=""
      actionArea={<FullAndBasicAccessOption selectedData={selectedData} />}
      buttonLabel="Save Permissions"
      nextCallback={() => changePage?.(2, true)}
      cancelCallback={() => dismissModal()}
    />,
    <SuccessSlide
      key="third"
      label="Level Of Access Saved"
      body={
        <Column className="items-center">
          <TextBox
            className="text-center"
            text="The Level of access to your account information has been updated for:"
          />
          <Spacer size={16} />
          <TextBox className="font-bold" text={memberName} />
        </Column>
      }
      doneCallBack={() => dismissModal()}
    />,
  ];

  return pages[pageIndex!];
};
