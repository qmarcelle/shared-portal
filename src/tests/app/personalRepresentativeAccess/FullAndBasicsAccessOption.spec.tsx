import FullAndBasicAccessOption from '@/app/(protected)/(common)/member/personalRepresentativeAccess/components/FullAndBasicAccessOption';
import { AccessType } from '@/app/(protected)/(common)/member/personalRepresentativeAccess/journeys/EditLevelOfAccess';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

const renderUI = (selectedValue: AccessType) =>
  render(
    <FullAndBasicAccessOption accessType={selectedValue} isMaturedMinor />,
  );
describe('Full And Basics Access Option', () => {
  it('should render the UI correctly for true', () => {
    const component = renderUI('basic');
    expect(screen.getByText('Grant Basic Access to Chris Hall'));
    expect(component.baseElement).toMatchSnapshot();
  });

  it('should render the UI correctly for false', () => {
    const component = renderUI('full');
    expect(
      screen.getByText(
        'HIPAA Authorization for Full Portal Access and Interoperability APIs',
      ),
    );
    expect(screen.getByText('Personal Representative Full Access'));
    expect(
      screen.getByText(
        'Demographic and contact information related to you and your family;',
      ),
    );
    expect(component.baseElement).toMatchSnapshot();
  });
});
