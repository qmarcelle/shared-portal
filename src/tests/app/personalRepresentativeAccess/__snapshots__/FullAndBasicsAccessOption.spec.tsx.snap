// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full And Basics Access Option should render the UI correctly for false 1`] = `
<body>
  <div>
    <main>
      <div
        class="flex flex-col items-center teritary-color-1"
      >
        <span
          class="rich-text body-1 undefined"
        >
          <span>
            Grant 
          </span>
          <span
            class="font-bold"
          >
            Full
             Access
             
          </span>
          <span>
            to 
          </span>
          <span
            class="font-bold"
          >
            Chris Hall
             
          </span>
        </span>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <div
          class=""
          style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
        />
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
        <span>
          <b>
            Member Name :
          </b>
           
          [Mature Minor Name]
        </span>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
        <div
          class="flex flex-row font-bold"
        >
          HIPAA Authorization for Full Portal Access and Interoperability
              APIs
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <span
          class="rich-text body-1 undefined"
        >
          <span>
            You are asking BlueCross BlueShield of Tennessee to give your 
          </span>
          <span
            class="link font-bold"
          >
            <a>
              Personal Representative Full Access 
            </a>
          </span>
          <span>
             
            to all of your health information available in your member portal and through our interoperability application programming interfaces (APIs), including our
          </span>
          <span
            class="link font-bold"
          >
            <a>
               Patient Access API 
            </a>
          </span>
          <span>
            and 
          </span>
          <span
            class="link font-bold"
          >
             
            <a>
               Payer-to-Payer API. 
            </a>
          </span>
          <span>
            This means that your Personal Representative will be able to access highly sensitive health information about you and to use our APIs to share your health information with other third parties through our APIs, such as any third-party application of your Personal Representative’s choosing or your other health plans.
          </span>
        </span>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <span
          class="rich-text body-1 undefined"
        >
          <span>
             By clicking the
          </span>
          <span
            class="font-bold"
          >
             "I Agree" 
          </span>
          <span>
            button after reading this, you authorize BlueCross BlueShield of
                Tennessee to share the following health information about you with your Personal Representative and any people, companies or
                other third parties authorized by your Personal Representative,if we maintain it:
          </span>
        </span>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <span
          class="rich-text body-1 undefined"
        >
          <span>
            <ul
              class="list-outside list-disc pl-5"
            >
              <li
                class="pb-2 marker:text-blue-500"
              >
                Demographic and contact information related to you and your family;
              </li>
              <li
                class="pb-2 marker:text-blue-500"
              >
                Your claims and encounter data, including cost information; and
              </li>
              <li
                class="pb-2 marker:text-blue-500"
              >
                Certain clinical data, such as allergies and intolerances, assessments and plans of treatment, care team members, clinical notes, clinical tests, diagnostic imaging, encounter information, goals, health insurance information, health status assessments, immunizations, laboratory tests and results, medications, problems, procedures, provenance, unique device identifiers for implantable devices, and vital signs.
              </li>
            </ul>
          </span>
        </span>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <span
          class="block"
        >
          The information they will have access to may reveal highly
                sensitive health information about you, including information
                about yours treatment or care for reproductive health (e.g.,
                family planning, contraception, miscarriage, abortion,
                maternity, infertility), substance use disorders (e.g., drugs
                and alcohol), mental or behavioral health disorders, HIV/AIDS,
                sexually transmitted diseases (STDs), communicable diseases,
                developmental or intellectual disabilities, genetic disorders,
                (including genetic testing for such disorders and genetic
                history), abuse (e.g., sexual, physical or mental), brain or
                other cognitive disorders, or other sensitive information. By
                clicking on the “I AGREE” button you are specifically
                authorizing the disclosure of this highly sensitive health
                information.
        </span>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <span
          class="rich-text body-1 undefined"
        >
          <span
            class="!block mb-4"
          >
            We are making this disclosure at your request, with your
                approval and at your direction. The 
                intended purposes of the use and disclosure of your health information, 
                including your highly sensitive health information, are to:
          </span>
          <span
            class="inline-block mb-4"
          >
            <ul
              class="list-disc list-outside pl-5"
            >
              <li
                class="pb-2 marker:text-blue-500"
              >
                Give you control over the use and redisclosure of your health information, including highly sensitive health information;
              </li>
              <li
                class="pb-2 marker:text-blue-500"
              >
                Give your Personal Representative the ability to use our interoperability APIs to disclose your highly sensitive health information to other people, companies and third parties authorized by your Personal Representative without needing further authorization from you to do so; and
              </li>
              <li
                class="pb-2 marker:text-blue-500"
              >
                Support treatment, payment and certain health care operations activities, including care coordination and case management among your care team, including family members, friends and others involved in your care.
              </li>
            </ul>
          </span>
          <span
            class="!block mb-4 pt-2"
          >
            You acknowledge that your Personal Representative
                 and the people, companies and other third parties that they authorize to receive your health information 
                may not be subject to state or
                federal health information privacy laws, such as the federal
                Health Information Portability and Accountability Act and its
                implementing regulations (collectively, called “HIPAA”) , 42
                U.S.C. § 290dd-2 and its implementing regulations at 42 CFR Part
                2 (collectively, called “Part 2”) for certain substance use
                disorder records, or state health information confidentiality
                laws. If your Personal Representative or someone that they have
                authorized to have access to your Part 2 substance use disorder
                records is a HIPAA covered entity or business associate
                (collectively, called a “HIPAA-regulated entity”) that is
                receiving your records to support treatment, payment or health
                care operations activities, the HIPAA-regulated entity recipient
                may redisclose those substance use disorder records for any
                purpose permitted by HIPAA, except for uses and disclosures for
                civil, criminal, administrative, and legislative proceedings
                against you.
          </span>
          <span
            class="inline-block pt-4"
          >
            This authorization for the release of your health information is voluntary. This authorization will remain in effect until one of the following events happen:
          </span>
          <span
            class="inline-block mb-4"
          >
            <ul
              class="list-outside list-disc pl-4 pt-4"
            >
              <li
                class="mb-2 marker:text-blue-500"
              >
                It is revoked (taken back).
              </li>
              <li
                class="mb-2 marker:text-blue-500"
              >
                XX calendar days after the last time you access your portal account. You understand that every time you access your portal account, you reaffirm this authorization and the xx-day clock starts again. If you do not access your portal account for more than xx-days, you will be asked to agree to a new authorization.
              </li>
              <li
                class="mb-2 marker:text-blue-500"
              >
                We reasonably determine that allowing 
                    the user you have
                    designated as your Personal Representative to link to yours 
                     account through our portal and APIs would present an
                    unacceptable security risk.
              </li>
              <li
                class="mb-2 marker:text-blue-500"
              >
                As otherwise provided for in this authorization or our User Agreement, such as if we learn that account credentials have been misused.
              </li>
            </ul>
          </span>
          <span
            class="mb-4 pt-4"
          >
            You understand that BlueCross BlueShield of Tennessee will not condition the provision of treatment, payment for treatment, enrollment in a health plan, or eligibility for benefits on your agreement to this authorization. You may take back this authorization in writing at any time, except to the extent the information has already been disclosed in reliance on this authorization. You may take back this authorization by sending a request to us the Contact Us page on bcbst.com or by sending us your written cancellation to BlueCross BlueShield of Tennessee, Privacy; 1 Cameron Hill Circle; Building 1, 5th Floor; Chattanooga, Tennessee 37402. If you decide to take back this authorization, it will not affect any disclosure BlueCross BlueShield made in reliance on this authorization before the cancellation.
          </span>
        </span>
        <span
          class="rich-text body-1 undefined"
        >
          <span
            class="!block mb-4 pt-4"
          >
            Please note that your Personal Representative will not have Full
                Access to your health information or the ability to use our APIs
                to disclose your information to other people, companies or third
                parties if you do not sign this authorization. If you do not
                sign this authorization, your Personal Representative will
                continue to have Basic Access to your health information in
                accordance with the laws that apply to BlueCross BlueShield of
                Tennessee.
          </span>
          <span
            class="font-bold"
          >
            Notice To Recipient of Substance Use Disorder Records:
          </span>
          <span>
             42 CFR Part 2 prohibits unauthorized use or disclosure of these
                records.
          </span>
        </span>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <div
          class=""
          style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
        />
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <div
          class="flex flex-row gap-2 p-2  checkbox-disabled "
        >
          <label>
            <input
              checked=""
              id="myCheckbox"
              name=""
              type="checkbox"
            />
          </label>
          <div
            class="flex flex-col"
          >
            <label
              for="myCheckbox"
            >
              <p />
            </label>
            <span
              class="rich-text body-1 undefined"
            >
              <span
                class="font-bold"
              >
                I Agree.
              </span>
              <span>
                 Clicking “I Agree” means that you have read, understand, and
                    agree to this authorization. You are giving your permission
                    for BlueCross BlueShield of Tennessee to give your Personal Representative, and any person, company or other third party
                    authorized by your Personal Representative, access to yours
                    health information, including your highly sensitive health
                    information. You understand that you can contact BlueCross
                    BlueShield of Tennessee for a copy of this authorization.
              </span>
            </span>
          </div>
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <div
          class="flex flex-col w-full text-field"
        >
          <p>
            Type Your Full Name
          </p>
          <div
            class="flex flex-row items-center input   "
          >
            <input
              aria-label="Type Your Full Name"
              class=""
              type="text"
              value=""
            />
            <div
              class="cursor-pointer"
            />
          </div>
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
      </div>
    </main>
  </div>
</body>
`;

exports[`Full And Basics Access Option should render the UI correctly for true 1`] = `
<body>
  <div>
    <main>
      <div
        class="flex flex-col items-center teritary-color-1"
      >
        <div
          class="flex flex-row body-1 text-center"
        >
          Grant Basic Access to Chris Hall
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
        />
        <div
          class=""
          style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
        />
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <div
          class="flex flex-row gap-2 p-2  checkbox-disabled "
        >
          <label>
            <input
              checked=""
              id="myCheckbox"
              name=""
              type="checkbox"
            />
          </label>
          <div
            class="flex flex-col"
          >
            <label
              for="myCheckbox"
            >
              <p />
            </label>
            I Agree. Lorem ipsum dolor sit amet, consectetur adipiscing elit.   Sed sodales, orci vitae ullamcorper dignissim,    ligula nulla condimentum lectus, feugiat interdum dui turpis eget nisl.    Duis at lectus vitae augue pulvinar dictum eu nec neque. Pellentesque maximus mattis justo id tincidunt.   Mauris porttitor
          </div>
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
        <div
          class="flex flex-col w-full text-field"
        >
          <p>
            Type Your Full Name
          </p>
          <div
            class="flex flex-row items-center input   "
          >
            <input
              aria-label="Type Your Full Name"
              class=""
              type="text"
              value="[Mature Minor Name]"
            />
            <div
              class="cursor-pointer"
            />
          </div>
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
      </div>
    </main>
  </div>
</body>
`;
